packages/sanity-plugin-nrkno-spellcheck/src/lib/core/words/cached-spellchecker.ts

```diff
@@ -0,0 +1,226 @@
+import uniq from 'lodash/uniq';
+import chunk from 'lodash/chunk';
+import {
+  Language,
+  SpellcheckJob,
+  SpellcheckProgress,
+  SpellcheckProgressCallback,
+  SpellcheckResult,
+  SpellcheckService,
+  SpellcheckTask,
+  WordMap,
+  WordSpelling,
+} from '../types';
+
+const DEFAULT_MAX_WORD_PER_REQUEST = 20;
+const DEFAULT_MAX_CONCURRENT_REQUESTS = 5;
+
+export interface CachedSpellcheckerConfig {
+  spellcheckService: SpellcheckService;
+  maxWordsPerRequest?: number;
+  maxConcurrentRequests?: number;
+}
+
+type WordCache = Map<string, WordSpelling>;
+
+interface SpellcheckJobState {
+  words: string[];
+  requestWords: string[];
+  failedWords: string[];
+  resolvedWords: WordMap;
+  progress: SpellcheckProgressCallback;
+  wordSuggestions: WordCache;
+}
+
+export class CachedSpellchecker {
```

Comment:
Refaktorert fra å være en funksjon med global state for cache, til klasse. Lettere å teste og man kan ha uavhengige cacheinstanser om man ønsker ved å ha flere instanser av klassen.
La også til støte for configurable wordparser.